dev/TeachingTip/TeachingTip.cpp

```diff
@@ -971,27 +975,58 @@ void TeachingTip::OnF6AcceleratorKeyClicked(const winrt::CoreDispatcher&, const
             m_hasF6BeenInvoked = true;
         }
 
-        winrt::Button f6Button = [this]() -> winrt::Button
+        auto const focusFromContent = [this, args]()
         {
-            auto firstButton = m_closeButton.get();
-            auto secondButton = m_alternateCloseButton.get();
-            if (CloseButtonContent())
+            auto parent = winrt::FocusManager::GetFocusedElement();
+            auto const rootElement = m_rootElement.get();
+            while (parent)
             {
-                std::swap(firstButton, secondButton);
+                if (auto parentAsUIElement = parent.try_as<winrt::UIElement>())
+                {
+                    if (parentAsUIElement == rootElement)
+                    {
+                        return true;
+                    }
+                }
+                parent = winrt::VisualTreeHelper::GetParent(parent.try_as<winrt::DependencyObject>());
             }
-            if (firstButton && firstButton.Visibility() == winrt::Visibility::Visible)
+            return false;
+        }();
+
+        if (focusFromContent)
+        {
+            if (auto&& previouslyFocusedElement = m_previouslyFocusedElement.get())
             {
-                return firstButton;
+                previouslyFocusedElement.Focus(winrt::FocusState::Keyboard);
             }
-            else if (secondButton && secondButton.Visibility() == winrt::Visibility::Visible)
+        }
+        else
+        {
+            const winrt::Button f6Button = [this]() -> winrt::Button
             {
-                return secondButton;
-            }
-            return nullptr;
-        }();
+                auto firstButton = m_closeButton.get();
+                auto secondButton = m_alternateCloseButton.get();
+                if (CloseButtonContent())
+                {
+                    std::swap(firstButton, secondButton);
+                }
+                if (firstButton && firstButton.Visibility() == winrt::Visibility::Visible)
```

Comment:
>firstButton.Visibility() == winrt::Visibility::Visible [](start = 35, length = 54)

Focusability requires more than just visibility - it also requires IsEnabled=true as one example.

Will these two buttons always be IsEnabled=true, and generally be focusable?

Reply:
Yes, modulo re-templating and other weird stuff that seems like paranoia to try to handle.

---
In reply to: [270931361](https://github.com/Microsoft/microsoft-ui-xaml/pull/511#discussion_r270931361) [](ancestors = 270931361)

Reply:
Sounds good.